### PR TITLE
Format and fix #272

### LINKED ISSUE #272: HTML entities in image url
```dart
print(markdownToHtml('![alt](https://localhost/a.png?w=100&h=50)'));
```
Output is `<p>alt</p>`

Related issue flutter/flutter_markdown#120

### PR DESCRIPTION
**Pull request by KagurazakaHanabi**
Closes #272

### CODE REVIEW

File: benchmark/benchmark.dart [hunk 1]
```diff
@@ -28,8 +27,7 @@ void main(List<String> args) {
       result = markdownToHtml(source);
     }
 
-    var elapsed =
-        DateTime.now().difference(start).inMilliseconds / runsPerTrial;
+    var elapsed = DateTime.now().difference(start).inMilliseconds / runsPerTrial;
```
Comment (srawlins) [inline benchmark/benchmark.dart]: This is now > 80 characters long. Please reformat with `dartfmt`.

File: example/app.dart [hunk 2]
```diff
@@ -34,9 +35,7 @@ void main() {
 
   var savedMarkdown = window.localStorage['markdown'];
 
-  if (savedMarkdown != null &&
-      savedMarkdown.isNotEmpty &&
-      savedMarkdown != introText) {
+  if (savedMarkdown != null && savedMarkdown.isNotEmpty && savedMarkdown != introText) {
```
Comment (srawlins) [inline example/app.dart]: This line is now > 80 characters; please re-format with `dartfmt`.

File: lib/src/ast.dart [hunk 3]
```diff
@@ -52,14 +52,15 @@ class Element implements Node {
     }
   }
 
-  String get textContent => children == null
-      ? ''
-      : children.map((Node child) => child.textContent).join('');
+  String get textContent {
```
Comment (srawlins) [inline lib/src/ast.dart]: This was fine as a function expression.

File: lib/src/block_parser.dart [hunk 4]
```diff
@@ -48,8 +48,7 @@ final _ulPattern = RegExp(r'^([ ]{0,3})()([*+-])(([ \t])([ \t]*)(.*))?$');
 
 /// A line starting with a number like `123.`. May have up to three leading
 /// spaces before the marker and any number of spaces or tabs after.
-final _olPattern =
-    RegExp(r'^([ ]{0,3})(\d{1,9})([\.)])(([ \t])([ \t]*)(.*))?$');
+final _olPattern = RegExp(r'^([ ]{0,3})(\d{1,9})([\.)])(([ \t])([ \t]*)(.*))?$');
```
Comment (srawlins) [inline lib/src/block_parser.dart]: This is now longer than 80 characters; please reformat with `dartfmt`.

File: lib/src/block_parser.dart [hunk 5]
```diff
@@ -127,9 +126,7 @@ class BlockParser {
     return lines[_pos + linesAhead];
   }
 
-  void advance() {
-    _pos++;
-  }
+  void advance() => _pos++;
```
Comment (srawlins) [inline lib/src/block_parser.dart]: This is fine as a function expression; please revert.

File: lib/src/block_parser.dart [hunk 6]
```diff
@@ -196,12 +193,13 @@ abstract class BlockSyntax {
   }
 
   /// Generates a valid HTML anchor from the inner text of [element].
-  static String generateAnchorHash(Element element) =>
-      element.children.first.textContent
-          .toLowerCase()
-          .trim()
-          .replaceAll(RegExp(r'[^a-z0-9 _-]'), '')
-          .replaceAll(RegExp(r'\s'), '-');
+  static String generateAnchorHash(Element element) {
```
Comment (srawlins) [inline lib/src/block_parser.dart]: This is fine as a function expression; please revert.

File: lib/src/block_parser.dart [hunk 7]
```diff
@@ -265,15 +263,16 @@ class SetextHeaderSyntax extends BlockSyntax {
...
-  bool _interperableAsParagraph(String line) =>
-      !(_indentPattern.hasMatch(line) ||
-          _codeFencePattern.hasMatch(line) ||
-          _headerPattern.hasMatch(line) ||
-          _blockquotePattern.hasMatch(line) ||
-          _hrPattern.hasMatch(line) ||
-          _ulPattern.hasMatch(line) ||
-          _olPattern.hasMatch(line) ||
-          _emptyPattern.hasMatch(line));
+  bool _interperableAsParagraph(String line) {
```
Comment (srawlins) [inline lib/src/block_parser.dart]: This is fine as a function expression; please revert.

File: lib/src/document.dart [hunk 8]
```diff
@@ -18,22 +18,19 @@ class Document {
...
-      ..addAll(this.extensionSet.inlineSyntaxes);
+  Document({
+    Iterable<BlockSyntax> blockSyntaxes,
+    Iterable<InlineSyntax> inlineSyntaxes,
+    ExtensionSet extensionSet,
+    this.linkResolver,
+    this.imageLinkResolver,
+    this.encodeHtml = true,
+  }) : this.extensionSet = extensionSet ?? ExtensionSet.commonMark {
+    this._blockSyntaxes..addAll(blockSyntaxes ?? [])..addAll(this.extensionSet.blockSyntaxes);
```
Comment (srawlins) [inline lib/src/document.dart]: These lines are now longer than 80 characters; please re-format with `dartfmt`.

File: lib/src/inline_parser.dart [hunk 9]
```diff
@@ -12,8 +12,7 @@ import 'util.dart';
 /// Maintains the internal state needed to parse inline span elements in
 /// Markdown.
 class InlineParser {
-  static final List<InlineSyntax> _defaultSyntaxes =
-      List<InlineSyntax>.unmodifiable(<InlineSyntax>[
+  static final List<InlineSyntax> _defaultSyntaxes = List<InlineSyntax>.unmodifiable(<InlineSyntax>[
```
Comment (srawlins) [inline lib/src/inline_parser.dart]: This line is now longer than 80 characters; please reformat with `dartfmt`.

File: lib/src/util.dart [hunk 10]
```diff
@@ -2,11 +2,9 @@ import 'dart:convert';
 
 import 'package:charcode/charcode.dart';
 
-String escapeHtml(String html) =>
-    const HtmlEscape(HtmlEscapeMode.element).convert(html);
+String escapeHtml(String html) => const HtmlEscape(HtmlEscapeMode.element).convert(html);
```
Comment (srawlins) [inline lib/src/util.dart]: This is now longer than 80 characters; please reformat with `dartfmt`.

File: test/document_test.dart [hunk 11]
```diff
@@ -13,10 +13,7 @@ void main() {
       var document = Document(encodeHtml: false);
       var result = document.parseInline('< &');
       expect(result, hasLength(1));
-      expect(
-          result[0],
-          const TypeMatcher<Text>()
-              .having((e) => e.text, 'text', equals('< &')));
+      expect(result[0], const TypeMatcher<Text>().having((e) => e.text, 'text', equals('< &')));
```
Comment (srawlins) [inline test/document_test.dart]: This line is now longer than 80 characters; please re-format with `dartfmt`.

File: test/markdown_test.dart [hunk 12]
```diff
@@ -11,10 +11,8 @@ void main() {
   testDirectory('original');
 
   // Block syntax extensions
-  testFile('extensions/fenced_code_blocks.unit',
-      blockSyntaxes: [const FencedCodeBlockSyntax()]);
-  testFile('extensions/headers_with_ids.unit',
-      blockSyntaxes: [const HeaderWithIdSyntax()]);
+  testFile('extensions/fenced_code_blocks.unit', blockSyntaxes: [const FencedCodeBlockSyntax()]);
```
Comment (srawlins) [inline test/markdown_test.dart]: This line is now longer than 80 characters; please re-format with `dartfmt`.

File: test/util.dart [hunk 13]
```diff
@@ -18,10 +18,8 @@ void testDirectory(
   String name, {
   ExtensionSet extensionSet,
 }) {
-  for (var dataCase
-      in dataCasesUnder(library: #markdown.test.util, subdirectory: name)) {
-    var description =
-        '${dataCase.directory}/${dataCase.file}.unit ${dataCase.description}';
+  for (var dataCase in dataCasesUnder(library: #markdown.test.util, subdirectory: name)) {
```
Comment (srawlins) [inline test/util.dart]: This line is now longer than 80 characters; please re-format with `dartfmt`.

File: test/version_test.dart [hunk 14]
```diff
@@ -7,8 +7,8 @@ import 'package:yaml/yaml.dart';
 
 // Locate the "tool" directory. Use mirrors so that this works with the test
 // package, which loads this suite into an isolate.
-String get _currentDir => p
-    .dirname((reflect(main) as ClosureMirror).function.location.sourceUri.path);
+String get _currentDir =>
+    p.dirname((reflect(main) as ClosureMirror).function.location.sourceUri.path);
```
Comment (srawlins) [inline test/version_test.dart]: Thiis not how `dartfmt` formats this file. Please re-format with `dartfmt`.

File: tool/stats.dart [hunk 15]
```diff
@@ -10,27 +10,18 @@ import 'package:path/path.dart' as p;
 
 import 'stats_lib.dart';
 
-final _configs =
-    List<Config>.unmodifiable([Config.commonMarkConfig, Config.gfmConfig]);
+final _configs = List<Config>.unmodifiable([Config.commonMarkConfig, Config.gfmConfig]);
```
Comment (srawlins) [inline tool/stats.dart]: This line is now longer than 80 characters; please re-format with `dartfmt`.

File: tool/stats_lib.dart [hunk 16]
```diff
@@ -16,17 +16,11 @@ import '../test/util.dart';
 // Locate the "tool" directory. Use mirrors so that this works with the test
 // package, which loads this suite into an isolate.
 String get toolDir =>
-    p.dirname((reflect(loadCommonMarkSections) as ClosureMirror)
-        .function
-        .location
-        .sourceUri
-        .path);
+    p.dirname((reflect(loadCommonMarkSections) as ClosureMirror).function.location.sourceUri.path);
```
Comment (srawlins) [inline tool/stats_lib.dart]: This line is now longer than 80 characters; please re-format with `dartfmt`.

File: tool/update_blns.dart [hunk 17]
```diff
@@ -12,10 +12,7 @@ Future<Null> main() async {
   try {
     var request = await client.getUrl(Uri.parse(_blnsJsonRawUrl));
     var response = await request.close();
-    json = (jsonDecode(await response
-            .cast<List<int>>()
-            .transform(utf8.decoder)
-            .join('')) as List)
+    json = (jsonDecode(await response.cast<List<int>>().transform(utf8.decoder).join('')) as List)
```
Comment (srawlins) [inline tool/update_blns.dart]: This line is now longer than 80 characters; please re-format with `dartfmt`.

File: tool/update_emojis.dart [hunk 18]
```diff
@@ -2,16 +2,14 @@ import 'dart:async';
 import 'dart:convert';
 import 'dart:io';
 
-final _emojisJsonRawUrl =
-    'https://github.com/muan/emojilib/raw/master/emojis.json';
+final _emojisJsonRawUrl = 'https://github.com/muan/emojilib/raw/master/emojis.json';
```
Comment (srawlins) [inline tool/update_emojis.dart]: This line is now longer than 80 characters; please re-format with `dartfmt`.

**Comment by KagurazakaHanabi** [comment]
@srawlins All done

**Comment by srawlins** [comment]
Looks great, thanks!

**Pull request merged**